Tutorial1 readme (#1204)
playmint/ds: contracts/src/maps/tutorial-room-1/README.md

> # Downstream Game Creation Tutorial 1
> ## Fabricators and CLI

Reviewer: should this first section be "aim" or "overview", something that can be consistent in each tutorial

```suggestion
# Downstream Game Creation Tutorial 1

## Aim
```

Author: like it

We will follow the steps below to create a simple downstream map, with your own tile layout and your own factory building.

Once complete, you will have used the core Downstream tools to create a map that looks like this when played:

<img src="./readme-images/screenshot.png" width=300>

> # Prerequisites

Reviewer: this heading and all subsequent should be H2s (there should only be one H1)

```suggestion
## Prerequisites
```

- This repository cloned to your desktop. (Instructions in the top [readme](../../../../README.md).)
- [Docker Desktop](https://docs.docker.com/get-docker/)
- [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)

# 1. Deploy the game

First, deploy an instance of Downstream locally using Docker.

> 
> - From a terminal in the repository root run:
> - `docker compose up --pull=always`
> 

Reviewer: might be good practice to use full code blocks wherever possible. This makes it look more like terminal entry, and allows picking a syntax.

````suggestion

From a terminal in the repository root run:

```bash
docker compose up --pull=always
```

````

Author: Much nicer.
Done all of these as a new commit as I rebased before I could take your code suggestions.

# 3. Create a map

There is no where for the Unit to go so we will now expand the world by creating some tiles.

First, open the the **tile-fabricator**, by opening a web browser and navigating to [http://localhost:3000/tile-fabricator].

Now, draw a map design:
- Left click each tile that you want on your map.
- Zoom in and out with mouse scroll.
- To remove a tile, select **UNDISCOVERED TILE** from the brush drop down and left click to remove tiles.
- Draw a map that looks like this:

<img src="./readme-images/step3.png" width=200>

Once you are happy, export the tiles manifest file:
- Select **Export**.
- Find the downloaded file. It will be named something like `20d6a4ed-bb78-4774-8c0a-50c28451a380.yml`.

Finally, create a new folder in your desktop. This folder is your 'map' and will contain manifests defining tiles, buildings, items and more. For now we'll just copy the new tiles in:
- Create a folder like /Tutorial1Map
- Copy the exported tiles manifest into the folder.
- Rename it to 'Locations.yaml'.

# 4. Deploy the new tiles

We will use the Downstream CLI to deploy our newly created tiles to our local Downstream instance.

> 
> First, install the CLI
> - From a terminal run
> - `npm i -g @playmint/ds-cli`
> - Check it is installed from any terminal folder by running
> - `ds help`
> 

Reviewer: ````suggestion

First, install the CLI. From a terminal run:

```bash
npm i -g @playmint/ds-cli
```

Check it is installed from any terminal folder by running:

```bash
ds help
```

````

Now copy your Burner private key from Downstream:
- Browse to [localhost:3000]([http://localhost:3000]).
- Click the player icon at the top left.

<img src="./readme-images/step4a.png" width=200>

- Click **show**

<img src="./readme-images/step4b.png" width=200>

- Highlight and copy the key.

> 
> Finally, run the command to deploy your map:
> - In a terminal at your new folder run the following
> - (where `<private-key>` can be pasted from the one you just copied)
> - `ds apply -n local -k <private-key> -f Locations.yaml`
> 

Reviewer: ````suggestion

Finally, run the command to deploy your map:

In a terminal at your new folder run the following, where `<private-key>` can be pasted from the one you just copied:

```bash
ds apply -n local -k <private-key> -f Locations.yaml
```

````

You should see the terminal output display a series of ✅s for each tile that is deployed.

<img src="./readme-images/step4c.png" width=200>

Browse to [localhost:3000]([http://localhost:3000]) and you should see your newly created map and be able to move your Unit around it!

<img src="./readme-images/step4d.png" width=200>

# 5. Create a new type of building

> The map is currently empty and there are no buildings to build. So we are going to create a new type of 'Factory' building that crafts a new kind of item.

Reviewer: try and drum the word Kind in there heh

```suggestion
The map is currently empty and there are no buildings to build. So we are going to create a new kind of 'Factory' building that crafts a new kind of item.
```

First, open the the **building-fabricator**, by opening a web browser and navigating to [http://localhost:3000/building-fabricator].

You should see a page like this:

<img src="./readme-images/step5a.png" width=200>

Now give the building a name and description.

Then, click on the **Output** tab and define the item that can be crafted by this factory:
- Select an icon.
- Choose a name.

You can change the appearance of the building by clicking on the arrows and colour squares.

<img src="./readme-images/step5b.png" width=200>

Once you are happy, export the building source code:
- Select **Export**.
- Find the downloaded zip file. It will be named something like `BasicFactory.zip`.

> - Extract the files to the same folder as tiles.yaml.

Reviewer 2: I think there isn't a `tiles.yaml` as one of the previous instructions were to rename the yaml to `Locations.yaml`
I think I read that correctly?

There are plenty for examples of pre-made building kinds in the ds repository that can be copy and pasted into your map folder.

We will add some decorative blocking buildings to your map and see how everything in the map folder can be deployed at once.

First copy the [blockers.yaml](./blockers.yaml) to your map folder.

Now, use the [tile-fabricator](http://localhost:3000/tile-fabricator) to add some blockers to your map:
- Press the **import** button, navigate to your map folder, and select blockers.yaml.
- New building types will now be available in the **Brush** drop done.

<img src="./readme-images/step7a.png" width=200>

- Choose a building type and place instances with left mouse click.

<img src="./readme-images/step7b.png" width=200>

- When you're happy, press the **export** button.
- Replace your old Locations.yaml with the new downloaded manifest.

# 8. Deploy the whole map folder

To complete the tutorial, we're going to deploy your whole map folder together.

First, stop the Downstream instance running in Docker with **ctrl-c**, and restart it by running `docker compose up`.

Once you see "ready", browse to the game and check you have a single tile world.

> Now you can apply the whole map folder by passing the -R command and a folder path **ds apply**
> - At a terminal In your map folder run:
> - `ds apply -n local -k <private-key> -R -f .`
> 

Reviewer: ````suggestion
Now you can `ds apply` the whole map folder by passing the `-R` flag and the folder path. At a terminal In your map folder run:

```bash
ds apply -n local -k <private-key> -R -f .
```

````
